src/Parsers/ASTIdentifier.cpp

```diff
@@ -16,59 +16,65 @@ namespace ErrorCodes
     extern const int SYNTAX_ERROR;
 }
 
+ASTIdentifier::ASTIdentifier(const String & short_name)
+    : full_name(short_name), name_parts{short_name}, semantic(std::make_shared<IdentifierSemanticImpl>())
+{
+    assert(!full_name.empty());
+}
 
-ASTPtr ASTIdentifier::clone() const
+ASTIdentifier::ASTIdentifier(std::vector<String> && name_parts_, bool special)
+    : name_parts(name_parts_), semantic(std::make_shared<IdentifierSemanticImpl>())
 {
-    auto ret = std::make_shared<ASTIdentifier>(*this);
-    ret->semantic = std::make_shared<IdentifierSemanticImpl>(*ret->semantic);
-    return ret;
+    assert(!name_parts.empty());
+    for (const auto & part : name_parts)
+    {
+        assert(!part.empty());
+        (void) part;  // otherwise not-used in release build
```

Comment:
Does `[[maybe_unused]]` work with `for`?

Reply:
Will check.